components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/DBHandler.java

```diff
@@ -0,0 +1,172 @@
+import java.sql.SQLException;
+
+
+/**
+ * This class represents key database operations related to worker data.
+ */
+
+public class DBHandler {
+    private static final Logger logger = LoggerFactory.getLogger(DBHandler.class);
+    private static DBHandler instance = new DBHandler();
+
+    private DBHandler() {
+
+    }
+
+    public static DBHandler getInstance() {
+        return instance;
+    }
+
+    /**
+     * Initialize the db connection.
+     *
+     * @throws RDBMSTableException data source cannot be found
+     */
+    Connection getConnection(HikariDataSource dataSource) {
+        try {
+            if (dataSource != null) {
+                Connection conn = dataSource.getConnection();
+                conn.setAutoCommit(true);
+                return conn;
+            } else {
+                throw new RDBMSTableException("Datasource Could not be found. " +
+                        "Hence cannot initialize the status dashboard.");
+            }
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Error initializing connection: " + e.getMessage(), e);
+        }
+    }
+
+
+    /**
+     * Insert worker data worker db.
+     *
+     * @return isSuccess
+     */
+    boolean insert(PreparedStatement stmt) {
+        try {
+            stmt.execute();
+            return true;
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Attempted execution of query [" + stmt.toString() + "] produced an " +
+                    "exceptions" +
+                    " in " + stmt.toString(), e);
+        } finally {
+            try {
+                stmt.close();
+            } catch (SQLException e) {
+                //ignore
+            }
+        }
+    }
+
+    /**
+     * Delete workers data to worker db.
+     *
+     * @return isSuccess.
+     */
+    public boolean delete(PreparedStatement stmt) {
+        try {
+            stmt.execute();
+            stmt.close();
+            return true;
+        } catch (SQLException e) {
+            throw new RDBMSTableException(e.getMessage() + " in " + stmt.toString(), e);
+        }
+    }
+
+    /**
+     * Select worker from the worker DB.
+     *
+     * @return list of object.
+     */
+
+    ResultSet select(PreparedStatement stmt) {
+        ResultSet rs;
+        try {
+            rs = stmt.executeQuery();
+        } catch (SQLException e) {
+            throw new RDBMSTableException("Error retrieving records from table '" + stmt.toString() + "': "
+                    + e.getMessage(), e);
+        }
+        return rs;
+    }
+
+    /**
+     * Update workers in the worker db.
+     *
+     * @return new object that updated;
+     */
+    Object update(PreparedStatement stmt) {
+        try {
+            stmt.execute();
+            stmt.close();
+            return stmt;
+        } catch (SQLException e) {
+            throw new RDBMSTableException(e.getMessage() + " in " + stmt.toString(), e);
```

Comment:
Make sure redundant details are not given through the exception message. No need to include e.getMessage() since it will be included in the stack trace.